pkg/kv/kvtest/iterators.go

```diff
@@ -0,0 +1,428 @@
+			t.Fatal("failed to set model", err)
+		}
+	}
+	err := sm.SetMsg(ctx, thirdPartitionKey, []byte("e"), &TestModel{})
+	if err != nil {
+		t.Fatal("failed to set model", err)
+	}
+
+	err = sm.SetMsg(ctx, thirdPartitionKey, []byte("f"), &TestModel{Name: []byte("notin")})
+	if err != nil {
+		t.Fatal("failed to set model", err)
+	}
+	// first partition - (a,a) (aa,aa) (b,b) (c,c) (d,d)
+	// second partition - (a,a) (aa,aa) (b,b) (c,c) (d,d)
+	// third partition - (e,nil) (f,"notin")
+
+	t.Run("listing all values of partition", func(t *testing.T) {
+		itr, err := kv.NewSecondaryIterator(ctx, store, (&TestModel{}).ProtoReflect().Type(),
+			firstPartitionKey, []byte(""), []byte(""))
+		if err != nil {
+			t.Fatalf("failed to create secondary iterator: %v", err)
+		}
+		defer itr.Close()
+		names := make([]string, 0)
+		for itr.Next() {
+			e := itr.Entry()
+			model, ok := e.Value.(*TestModel)
+			if !ok {
+				t.Fatalf("cannot read from store")
+			}
+			names = append(names, string(model.Name))
+		}
+		if itr.Err() != nil {
+			t.Fatalf("unexpected error: %v", itr.Err())
+		}
+
+		if diffs := deep.Equal(names, []string{"a", "aa", "b", "c", "d"}); diffs != nil {
+			t.Fatalf("got wrong list of names: %v", diffs)
+		}
+	})
+
+	t.Run("listing with prefix", func(t *testing.T) {
+		itr, err := kv.NewSecondaryIterator(ctx, store, (&TestModel{}).ProtoReflect().Type(),
+			firstPartitionKey, []byte("a"), []byte(""))
+		if err != nil {
+			t.Fatalf("failed to create secondary iterator: %v", err)
+		}
+		defer itr.Close()
+		names := make([]string, 0)
+		for itr.Next() {
+			e := itr.Entry()
+			model, ok := e.Value.(*TestModel)
+			if !ok {
+				t.Fatalf("cannot read from store")
+			}
+			names = append(names, string(model.Name))
+		}
+		if itr.Err() != nil {
+			t.Fatalf("unexpected error: %v", itr.Err())
+		}
+
+		if diffs := deep.Equal(names, []string{"a", "aa"}); diffs != nil {
+			t.Fatalf("got wrong list of names: %v", diffs)
+		}
+	})
+
+	t.Run("listing after", func(t *testing.T) {
+		itr, err := kv.NewSecondaryIterator(ctx, store, (&TestModel{}).ProtoReflect().Type(),
+			firstPartitionKey, []byte(""), []byte("b"))
+		if err != nil {
+			t.Fatalf("failed to create secondary iterator: %v", err)
+		}
+		defer itr.Close()
+		names := make([]string, 0)
+		for itr.Next() {
+			e := itr.Entry()
+			model, ok := e.Value.(*TestModel)
+			if !ok {
+				t.Fatalf("cannot read from store")
+			}
+			names = append(names, string(model.Name))
+		}
+		if itr.Err() != nil {
+			t.Fatalf("unexpected error: %v", itr.Err())
+		}
+
+		if diffs := deep.Equal(names, []string{"c", "d"}); diffs != nil {
+			t.Fatalf("got wrong list of names: %v", diffs)
+		}
+	})
+
+	t.Run("entry not found", func(t *testing.T) {
+		itr, err := kv.NewSecondaryIterator(ctx, store, (&TestModel{}).ProtoReflect().Type(),
+			thirdPartitionKey, []byte(""), []byte(""))
+		if err != nil {
+			t.Fatalf("failed to create secondary iterator: %v", err)
+		}
+		defer itr.Close()
+
+		for itr.Next() {
```

Comment:
in case we expect an error we should test Next() returns false

Reply:
It returns false because it will exit the loop

Reply:
True, so we need to verify it returns false and delete a code that never should run.